website/project/views/node.py

```diff
@@ -1054,18 +1052,16 @@ def search_node(auth, **kwargs):
     if not query:
         return {'nodes': []}
 
-    # Build ODM query
-    title_query = Q('title', 'icontains', query)
-    not_deleted_query = Q('is_deleted', 'eq', False)
-    visibility_query = Q('contributors', 'eq', auth.user)
-    if include_public:
-        visibility_query = visibility_query | Q('is_public', 'eq', True)
-    odm_query = title_query & not_deleted_query & visibility_query
-
     # Exclude current node from query if provided
     nin = [node.id] + list(node._nodes.values_list('pk', flat=True)) if node else []
 
-    nodes = AbstractNode.find(odm_query).exclude(id__in=nin).exclude(type='osf.collection')
+    nodes = AbstractNode.objects.filter(
+        Q(contributors=auth.user) |
+        Q(is_public=bool(include_public)),
+        title__icontains=query,
+        is_deleted=False
+    ).exclude(id__in=nin, type='osf.collection')
```

Comment:
Is this logically equivalent to the old code?

```python
exclude(id__in=nin).exclude(type='osf.collection')
```

should exclude nodes whose `id` is in `nin` and exclude collections.

I think that

```python
.exclude(id__in=nin, type='osf.collection')
```

will only exclude collections whose `id` is in `nin` .

Reply:
Definitely not equivalent. Nice catch.

```
WHERE ("osf_abstractnode"."type" = osf.node AND NOT ("osf_abstractnode"."is_public" = True AND "osf_abstractnode"."type" = osf.collection))
```
vs
```
WHERE ("osf_abstractnode"."type" = osf.node AND NOT ("osf_abstractnode"."is_public" = True) AND NOT ("osf_abstractnode"."type" = osf.collection))
```